courses/models.py

```diff
@@ -290,6 +290,13 @@ class CourseTopic(TimestampedModel):
     """
 
     name = models.CharField(max_length=128, unique=True)
+    parent = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
```

Comment:
Could we add a more useful message for the user while deleting topics? Maybe we should mention that it might delete child courses.
<img width="886" alt="Screen Shot 2023-03-28 at 4 48 34 PM" src="https://user-images.githubusercontent.com/52656433/228226671-060bb8b6-6554-4eaf-b2ee-e8c295ff144f.png">

Reply:
Done in https://github.com/mitodl/mitxpro/pull/2596/commits/ceb557cfb593b48ddb98e4a6d4188fa7f4b644c8